cmd/api/src/daemons/datapipe/decoders.go

```diff
@@ -56,17 +57,22 @@ func decodeBasicData[T any](batch graph.Batch, reader io.ReadSeeker, conversionF
 		}
 
 		if count == IngestCountThreshold {
-			IngestBasicData(batch, convertedData)
+			if err = IngestBasicData(batch, convertedData); err != nil {
+				errs = append(errs, err)
+			}
 			convertedData.Clear()
 			count = 0
+
 		}
 	}
 
 	if count > 0 {
-		IngestBasicData(batch, convertedData)
+		if err = IngestBasicData(batch, convertedData); err != nil {
+			errs = append(errs, err)
+		}
 	}
 
-	return nil
+	return errors.Join(errs...)
```

Comment:
`errors.Join()` is great conceptually, but I think we have been avoiding using it because it does not play nice with the logging due to newlines? We have a custom compound error structure we've been using instead sometimes